urllib3/connectionpool.py

```diff
@@ -687,12 +687,17 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                 retries = retries.increment(method, url, response=response, _pool=self)
             except MaxRetryError:
                 if retries.raise_on_redirect:
-                    # Release the connection for this response, since we're not
-                    # returning it to be released manually.
+                    # Drain and release the connection for this response, since
+                    # we're not returning it to be released manually.
+                    response.read()
                     response.release_conn()
```

Comment:
Cool, so with all that code duplication about it'd be nice to pull this out to a handy-dandy helper function. Declaring it inline in `urlopen` would probably be enough.